packages/admin/admin/src/common/toolbar/DataGridToolbar.tsx

```diff
@@ -55,8 +57,11 @@ const Root = createComponentSlot(Toolbar)<DataGridToolbarClassKey, OwnerState>({
 export const DataGridToolbar = (inProps: DataGridToolbarProps) => {
     const { density = "standard", slotProps, ...restProps } = useThemeProps({ props: inProps, name: "CometAdminDataGridToolbar" });
 
+    const apiRef = useGridApiContext();
+    const gridDensity = apiRef.current?.state?.density.value;
+
     const ownerState: OwnerState = {
-        density,
+        density: gridDensity || density,
```

Comment:
Shouldn't we first use the provided prop and then the density of the data grid? Otherwise the prop will never be used I guess. @jamesricky what do you think?

Reply:
Yes, if the `density` prop is set, it should be used and it should not have a default-value. 
If the prop is not set, the `gridDensity` should be used. 

Sorry, my explanation in the Task on this was a bit confusing.